lib/atc/utils/file_utils.rb

```diff
@@ -1,6 +1,11 @@
 # frozen_string_literal: true
 
 module Atc::Utils::FileUtils
+  def self.source_object_for_path(local_file_path)
```

Comment:
How would you feel about adding this find-by-path functionality to the SourceObject model itself?